backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -56,33 +58,48 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Request parameter values
-    // TODO: Check input values are filled (before allowing to submit)
-    UserService userService = UserServiceFactory.getUserService();
-    String email = userService.getCurrentUser().getEmail();
-    if (email == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
-      return;
+
+    // Retrieve body of HTTP request
+    StringBuffer requestBuffer = new StringBuffer();
+    try {
+      BufferedReader reader = request.getReader();
+      String currentLine;
+      while ((currentLine = reader.readLine()) != null) {
+        requestBuffer.append(currentLine);
+      }
+    } catch (Exception e) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Could not read request body");
     }
-    String username = email.split("@")[0];
+    JSONObject obj = new JSONObject(requestBuffer.toString());
+
+    // Retrieve the timeAvailableUntil input and convert to a UTC ZonedDateTime
+    long timeAvailableUntil = obj.getJSONObject("formDetails").getLong("timeAvailableUntil");
```

Comment:
*endTimeAvailable. Also, how is this input a long if the user only input a time without a date?

Reply:
When I send it values over from the frontend, I use `getTime()` which allows me to retrieve the long

Reply:
See my other comment below about getTime() on the frontend